hasher-matcher-actioner/hmalib/common/count_models.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import typing as t
+from decimal import Decimal
+
+from mypy_boto3_dynamodb.service_resource import Table
+from boto3.dynamodb.conditions import Attr
+
+
+class MatchByPrivacyGroupCounter:
+    """
+    Updates and retrieves counts of matches by Privacy Group.
+
+    Everything is a class method so you can extend this class, override what you
+    see fit and get the behavior without writing more than you need to. This
+    class might in the future become a base class for all counters.
+    """
+
+    @classmethod
+    def get_all_counts(cls, table: Table) -> t.Dict[str, int]:
+        response = table.scan(FilterExpression=Attr("PK").eq(cls._get_pkey()))
```

Comment:
[Diff 2/5] scan could be unnecessarily pricey if we store this in the primary datastore. If we know can look up the SKs we want before had we should probably use them even if we want all counts... 
At the very least we could add a filter based on SK's prefix...

Reply:
I do not think it is pricey if the PK is the same. It will go to the same partition so the DTUs utilized should be limited.